BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/BuildProperties.groovy

```diff
@@ -0,0 +1,41 @@
+package com.novoda.buildproperties
+
+class BuildProperties {
+
+    final String name
+    File file
+    Properties entries
+
+    BuildProperties(String name) {
+        this.name = name
+    }
+
+    String getName() {
```

Comment:
But you have setFile and setEntries as well.  Aren't those getter automatically available to the consumer even when they are not defined explicitly.